client/client.go

```diff
@@ -120,14 +123,18 @@ func (c *Client) Dial(prm PrmDial) error {
 	}
 
 	// TODO: (neofs-api-go#382) perform generic dial stage of the client.Client
-	_, err := rpc.Balance(&c.c, new(v2accounting.BalanceRequest),
-		client.WithContext(prm.parentCtx),
-	)
+	endpointInfo, err := c.EndpointInfo(prm.parentCtx, PrmEndpointInfo{})
 	// return context errors since they signal about dial problem
 	if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
 		return err
 	}
 
+	if endpointInfo == nil && err != nil {
```

Comment:
This effectively means returning any error. I don't mind, but then the check for `endpointInfo` is not really needed.

Reply:
OK, I suggest to left here all errors returning, because we got panic here, if `endpointInfo` is nil